plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -49,14 +50,24 @@ var ErrNoNodesAvailable = fmt.Errorf("no nodes available to schedule pods")
 func (f *FitError) Error() string {
 	var buf bytes.Buffer
 	buf.WriteString(fmt.Sprintf("pod (%s) failed to fit in any node\n", f.Pod.Name))
-	for node, predicates := range f.FailedPredicates {
-		reasons := make([]string, 0)
+	reasons := maps.NewCounter()
```

Comment:
From this use... you don't need it to be thread safe?

So just make this map[string]int{} and manage it yourself and remove the utility package.